sycl/include/sycl/interop_handle.hpp

```diff
@@ -70,6 +71,31 @@ class interop_handle {
 #endif
   }
 
+  /// Receives a SYCL accessor that has been defined as a requirement for the
+  /// command group, and returns the underlying OpenCL memory object that is
+  /// used by the SYCL runtime. If the accessor passed as parameter is not part
+  /// of the command group requirements (e.g. it is an unregistered placeholder
+  /// accessor), the exception `sycl::invalid_object` is thrown
+  /// asynchronously.
+  template <backend Backend = backend::opencl, typename DataT, int Dims,
+            access::mode Mode, access::target Target, access::placeholder IsPlh>
+  backend_return_t<Backend, image<Dims>> get_native_mem(
+      const detail::image_accessor<DataT, Dims, Mode,
+                                   /*access::target::image*/ Target,
+                                   IsPlh /*, PropertyListT */> &Acc) const {
```

Comment:
This overload confuses me a little. `detail::image_accessor` is a base-class of other accessor. Why can't we just have the one above, but with the inverse `enable_if` condition?

Reply:
It could be done that way too. I thought this was clearer because accessor is complex.   We have `image_accessor` but there is no `buffer_accessor`, so the first specialization is for any type of accessor when not using target image, and this one is for `image_accessor`.   Meaning, if in the future we were to add a third or fourth type of accessor, this code will continue to work without needing an update.

Reply:
> Meaning, if in the future we were to add a third or fourth type of accessor, this code will continue to work without needing an update.

You mean if we add more accessors variants that inherit from `image_accessor`? If so, I worry that this implicit behavior would be as much a detriment as suddenly we have a potentially undocumented `get_native_mem` variant.

As for how clear this is, I somewhat agree that using something called `image_accessor` is clearer (and with SYCL 2020 image accessors will be clearer to the user too), but while reading it I saw the `detail` namespace and my first impression was that this was an internal interface in the public space.

Reply:
I don't believe this is a concern big enough to block this PR.